xcell/mappers/mapper_HSC_DR1wl.py

```diff
@@ -0,0 +1,278 @@
+                return self.cat
+
+            cats = []
+            for f in self.config['data_catalogs']:
+                cat = self._clean_raw_catalog(f)
+
+                # Shear cut
+                shear_mod_thr = self.config.get('shear_mod_thr', 2)
+                isn = 'ishape_hsm_regauss'
+                ishape_flags_mask = ~cat[f'{isn}_flags']
+                ishape_sigma_mask = ~np.isnan(cat[f'{isn}_sigma'])
+                ishape_res_mask = cat[f'{isn}_resolution'] >= 0.3
+                ishape_shear_mod_mask = (cat[f'{isn}_e1']**2 +
+                                         cat[f'{isn}_e2']**2) < shear_mod_thr
+                ishape_sigma_mask *= ((cat[f'{isn}_sigma'] >= 0.) *
+                                      (cat[f'{isn}_sigma'] <= 0.4))
+                # Remove masked objects
+                imsk = 'iflags_pixel_bright'
+                star_mask = np.logical_not(cat[f'{imsk}_object_center'])
+                star_mask *= np.logical_not(cat[f'{imsk}_object_any'])
+                fdfc_mask = cat['wl_fulldepth_fullcolor']
+                shearmask = ishape_flags_mask *\
+                    ishape_sigma_mask *\
+                    ishape_res_mask *\
+                    ishape_shear_mod_mask*star_mask *\
+                    fdfc_mask
+                cat = cat[shearmask]
+
+                # Redshift bin cut
+                zs = cat['pz_best_eab']
+                zbinmask = (zs <= self.z_edges[1]) & (zs > self.z_edges[0])
+                cat = cat[zbinmask]
+
+                # Calibrate shear
+                w = cat[self.w_name]
+                mhat = np.average(cat[f'{isn}_derived_shear_bias_m'],
+                                  weights=w)
+                resp = 1. - np.average(cat[f'{isn}_derived_rms_e'] ** 2,
+                                       weights=w)
+                e1 = (cat[f'{isn}_e1']/(2.*resp) -
+                      cat[f'{isn}_derived_shear_bias_c1']) / (1 + mhat)
+                e2 = (cat[f'{isn}_e2']/(2.*resp) -
+                      cat[f'{isn}_derived_shear_bias_c2']) / (1 + mhat)
+                cat['e1'] = e1
+                cat['e2'] = e2
+
+                # Remove unnecessary columns
+                cat.keep_columns(['ra', 'dec', 'e1', 'e2', self.w_name])
+                cats.append(cat)
+            self.cat = vstack(cats)
+            if fname_lite is not None:
+                self.cat.write(fname_lite)
+        return self.cat
+
+    def _clean_raw_catalog(self, fnames):
+        cats = []
+        for fname in fnames:
+            if not os.path.isfile(fname):
+                raise ValueError(f"File {fname} not found")
+            c = Table.read(fname)
+            sel = np.ones(len(c), dtype=bool)
+            isnull_names = []
+            for key in c.keys():
+                if key.__contains__('isnull'):
+                    if not key.startswith('ishape'):
+                        sel[c[key]] = 0
```

Comment:
`sel` is of bool type. Shouldn't it be set to `False`? Save for L. 111

Reply:
0 is the equivalent of False if the array is already of type bool (which it is)